build: Add Melos and update contributing doc (#1207)
flame-engine/flame: .github/pull_request_template.md

# Description

*Replace this paragraph with a description of what this PR is doing. 
If you're modifying existing behavior, describe the existing behavior, how this PR is changing it,
and what motivated the change. If this is a breaking change, specify explicitly which APIs have been
changed.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes tests for ALL new/updated/fixed functionality.
- [ ] My PR does not decrease the code coverage.
- [ ] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `doc:` etc).
- [ ] I updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [ ] I updated/added relevant examples in `examples`.

> - [ ] I have formatted my code with `flutter format` and `flutter analyze` does not report any problems.

Reviewer: should we just replace this build is green to account for format, analyze, docs, tests, etc

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.

> - [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).

Reviewer 2: ```suggestion
- [ ] My PR includes tests for ALL new/updated/fixed functionality.
```

> - [ ] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.

Reviewer 2: ```suggestion
- [ ] My PR does not decrease the code coverage.
```
(if anyone wants to decrease coverage, they can keep this unchecked and then explain in the PR).

> - [ ] My PR title starts with a [Conventional Commit] type (`fix:`, `feat:`, `doc:` etc).

Reviewer 2: ```suggestion
- [ ] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `doc:` etc).
```
Previous items have "my PR" as a subject, so better to put forward the word **title** here.

> - [ ] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `examples`.

Reviewer 2: There are doc-comments, and then there are also docs in the `docs` folder.
I would put examples into a separate checklist item.

- [ ] I have formatted my code with `flutter format` and `flutter analyze` does not report any problems.

> - [ ] I read and followed the [Flame Style Guide].

Reviewer 2: Probably not needed, since this is covered by the [Contributor Guide] part.
